lib/std/target.zig

```diff
@@ -61,7 +61,7 @@ pub const Target = struct {
                 };
             }
 
-            pub fn dynamicLibSuffix(tag: Tag) [:0]const u8 {
+            pub fn dynamicLibSuffix(tag: Tag) []const u8 {
```

Comment:
This and `oFileExt`, and others that return infinite lifetimes should keep their sentinel terminated slice return type, for the same reason as string literals.

Reply:
> same reason as string literals.

What is that reason?

Reply:
See #265

Reply:
I don't see a rationale in there for all strings to be null-terminated?

Infact I see:
> We want to steer users away from this type and instead use []const u8, which includes a pointer and a length.

Reply:
Sometimes null termination is desirable for various reasons, such as interacting with an operating system ABI. Being forced to go from no memory management to memory management due to copying a string and adding a sentinel is a high cost. Putting a null byte at the end of a string literal in the data section is a low cost.